bot.py

```diff
@@ -906,56 +894,55 @@ async def onVote(self, ctx):
+    # For debugging, use the 3-second loop and the 10-second delta below
+    #@tasks.loop(seconds=3)
     @tasks.loop(hours=8)
     async def cleanupInactiveGames(self):
         print("Checking for inactive games")
         # Inactive games are ones that haven't had any activity in the last 3 hours
         inactiveTime = datetime.datetime.now() - datetime.timedelta(hours=3)
+        #inactiveTime = datetime.datetime.now() - datetime.timedelta(seconds=10)
 
         numEnded = 0
 
-        townsToEndByGuild = {}
+        # "Range query" that gives us all documents with a timestamp less than our inactive time
+        inactiveQuery = {"lastActivity" : {"$lt": inactiveTime}}
         
-        # TODO: This could use the DB instead of a cog member
-
-        # Collect all the inactive towns
-        for g, go in self.currentlyActiveGames.items():
-            townsToEndByGuild[g] = []
-            for t, to in go.items():
-                if "lastActivity" not in to or to["lastActivity"] < inactiveTime:
-                    townsToEndByGuild[g].append(t)
-
-        # Actually run endGame on each inactive town
-        for g, ts in townsToEndByGuild.items():
-            guild = self.bot.get_guild(g)
-            if guild:
-                for t in ts:
-                    query = { "guild" : guild.id, "controlChannelId" : t }
-                    doc = guildInfo.find_one(query)
-                    if doc:
-                        info = TownInfo(guild, doc)
-                        try:
-                            await self.onEndGameInternal(guild, info)
-                            numEnded = numEnded + 1
-                        except Exception:
-                            pass
+        for rec in g_dbActiveGames.find(inactiveQuery):
+            guildId = rec["guild"]
+            guild = self.bot.get_guild(guildId)
+            # Fetch the full info for this guild/channel combo
+            lookupQuery = { "guild" : guildId, "controlChannelId" : rec["channel"] }
+            doc = g_dbGuildInfo.find_one(lookupQuery)
+            if doc:
+                info = TownInfo(guild, doc)
+                try:
+                    await self.onEndGameInternal(guild, info)
+                    numEnded = numEnded + 1
+                except Exception:
+                    pass
 
         if numEnded > 0:
-            print("Ended " + str(numEnded) + " inactive game(s)")
+            print(f'Ended {numEnded} inactive game(s)')
 
         # Remove all the inactive towns from our list
-        for g, ts in townsToEndByGuild.items():
-            for t in ts:
-                self.currentlyActiveGames[g].pop(t, None)
-            if not self.currentlyActiveGames[g]:
-                self.currentlyActiveGames.pop(g, None)
+        result = g_dbActiveGames.delete_many(inactiveQuery)
 
         # If nothing is left active at all, stop the timer
-        if not self.currentlyActiveGames:
+        if g_dbActiveGames.count_documents({}) == 0:
```

Comment:
Is there a way to do an `any()` rather than a `count()>0`? We don't care how many there are, just that there are any.